### PR TITLE
Feature/user/register

### PR DESCRIPTION
**Pull request by DobocanDarius**

### CODE REVIEW

File: InternshipProject#2/Helpers/PasswordHash.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+
+namespace InternshipProject_2.Helpers;
+
+public class PasswordHash
```
Comment (iiusan) [inline InternshipProject#2/Helpers/PasswordHash.cs:6]: needs a better name

Comment (iiusan) [reply InternshipProject#2/Helpers/PasswordHash.cs:6]: PasswordHash.HashPassword() makes no sense

File: RequestResponseModels/User/Request/CreateUserRequest.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public CreateUserRequest(string username, string password, string email, string role)
+    {
+        Username = username;
+        Password = password;
+        Email = email;
+        Role = role;
+        CreatedAt = DateTime.Now;
+    }
+    public string Username { get; set; } = null!;
```
Comment (iiusan) [inline RequestResponseModels/User/Request/CreateUserRequest.cs:13]: remove null!

**Review by iiusan — CHANGES_REQUESTED** [review]

File: InternshipProject#2/Helpers/AutoMapper.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace AutoMapper
+{
+    public class MapperConfig
+    {
+        public static Mapper InitializeAutomapper()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+
```
Comment (iiusan) [inline InternshipProject#2/Helpers/AutoMapper.cs:20]: empty line

File: InternshipProject#2/Helpers/GenerateToken.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<<<<<<<< HEAD:InternshipProject#2/Helpers/GenerateToken.cs
+using InternshipProject_2.Models;
+========
+﻿using InternshipProject_2.Models;
+>>>>>>>> master:InternshipProject#2/Helpers/TokenGenerator.cs
```
Comment (iiusan) [inline InternshipProject#2/Helpers/GenerateToken.cs:5]: what is this ????
you didn't resolve the conflicts....

File: InternshipProject#2/Helpers/TokenGenerator.cs [hunk 5]
```diff
@@ -1,15 +1,27 @@
...
+========
 ﻿using InternshipProject_2.Models;
+>>>>>>>> master:InternshipProject#2/Helpers/TokenGenerator.cs
 using Microsoft.IdentityModel.Tokens;
 using System.IdentityModel.Tokens.Jwt;
 using System.Security.Claims;
 
 namespace InternshipProject_2.Helpers;
 
+<<<<<<<< HEAD:InternshipProject#2/Helpers/GenerateToken.cs
```
Comment (iiusan) [inline InternshipProject#2/Helpers/TokenGenerator.cs:12]: this code isn't even compilable.

**Review by iiusan — CHANGES_REQUESTED** [review]
This PR is very messy. You need to be more careful.

What is here isn't even ready for a review.

File: InternshipProject#2/Helpers/PasswordHash.cs (hunk 1 above)
Comment (DobocanDarius) [reply InternshipProject#2/Helpers/PasswordHash.cs]: changed name

File: InternshipProject#2/Helpers/AutoMapper.cs (hunk 3 above)
Comment (DobocanDarius) [reply InternshipProject#2/Helpers/AutoMapper.cs]: fixed line

File: InternshipProject#2/Helpers/GenerateToken.cs (hunk 4 above)
Comment (DobocanDarius) [reply InternshipProject#2/Helpers/GenerateToken.cs]: fixed conflict

File: InternshipProject#2/Helpers/TokenGenerator.cs (hunk 5 above)
Comment (DobocanDarius) [reply InternshipProject#2/Helpers/TokenGenerator.cs]: fixed conflict

File: InternshipProject#2.sln [hunk 6]
```diff
@@ -17,6 +20,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "RequestResponseModels", "Re
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "RequestResponseModels", "RequestResponseModels\RequestResponseModels.csproj", "{E44B5B0F-2642-42DD-A991-9689D9EF65A8}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "UnitTests", "UnitTests\UnitTests.csproj", "{814FA560-DA02-40C5-ACAF-C18D2C5D8F0C}"
+
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RequestResponseModels", "RequestResponseModels\RequestResponseModels.csproj", "{04BEB576-E0A0-452F-AF2C-478475F14FD3}"
```
Comment (iiusan) [inline InternshipProject#2.sln:24]: You should confirm this is correct....

Comment (iiusan) [reply InternshipProject#2.sln:24]: You have 2 references to the same project

Comment (DobocanDarius) [reply InternshipProject#2.sln]: deleted all additional references to the same projects

**Review by iiusan — APPROVED** [review]

**Pull request merged**